packages/@okta/vuepress-site/guides/sign-into-native-ui/primary-authentication/iOS/primary-authentication.md

```diff
@@ -0,0 +1,29 @@
+### Building UI
+
+There are different techniques how to build UI flows in iOS application. You can choose from MVVM, Viper or just use massive view controllers. Managing transitions between view controllers is also up to you, however we recommend to use [flow coordinator pattern](https://medium.com/@dkw5877/flow-coordinators-333ed64f3dd). Your flow coordinator instance will handle all the status updates and manage UI navigation. We've also implemented this pattern in our [sample application](https://github.com/okta/samples-ios/tree/master/custom-sign-in)
+
+As an example your application could have the following view controllers:
+- `LoginViewController` - responsible for reading username and password information from a user
+- `PasswordChangeViewController` - responsible for changing password or skip actions
+- `MFARequiredViewController` - responsible for showing user the list of available factors to choose from
+- `MFAChallengeViewController` - responsible for factor verification and retry actions
+- and etc.
+
+### Primary authentication
+
+With the primary authentication flow(no MFA) you typically need two view controllers  `LoginViewController` and `PasswordChangeViewController`.
```

Comment:
Let's remove password change from scope for now. By following this tutorial, the developer will have an app that does primary authentication, and shows an error screen for any response other than `SUCCESS`.